### ISSUE TITLE
Wrong router service used in PaymentController

### ISSUE THREAD

**Issue by AndreasA** [mollie/Shopware6#244]
It seems that in the PaymentController the router service https://github.com/mollie/Shopware6/blob/master/src/Resources/config/services.xml#L117

shouldn't it just be "router" so that storefront domain specific URLs are built (which includes local, currency and sales channel information).

Only relevant in case of using the mollie specific failed payment method handling instead of using the Shopware failed payment method handling.

**Comment by boxblinkracer** [comment]
Hi @AndreasA 
thanks for this

i double checked with the apple pay controller i did, and there is also the plain "router" being used.
so i think the sport you found might just be an accident.
i'll double check with the dev and change it :)

**Comment by boxblinkracer** [comment]
mhm, i just saw that was on purpose in a git commit
so i have to check why this was done

**Comment by boxblinkracer** [comment]
ok so from the ticket i see
that the symfony router was used with "router" and had to be changed to the shopware router which seems to be "router.default", to work correctly with the sales channel base urls.

so just like what you mentioned, but inverted? 
is there an error on your side, or was this just a question?

**Comment by AndreasA** [comment]
Should be inverted. As Shopware adds the Symfony router as `router.default` and decorate the original router which is `router`. So the Storefront router is available as `router`.
And it is important to use that one as there might not even be a sales channel added to the `/` router of the website which would result in a `SalesChannelMappingException`.

You can also check:

```shell
php7.4 bin/console debug:container router.default

Information for Service "router.default"
========================================

 This Router creates the Loader only when the cache is empty.

 ---------------- --------------------------------------------------- 
  Option           Value                                              
 ---------------- --------------------------------------------------- 
  Service ID       router.default                                     
  Class            Symfony\Bundle\FrameworkBundle\Routing\Router      
  Tags             monolog.logger (channel: router)                   
                   container.service_subscriber (id: routing.loader)  
  Calls            setConfigCacheFactory                              
  Public           no                                                 
  Synthetic        no                                                 
  Lazy             no                                                 
  Shared           yes                                                
  Abstract         no                                                 
  Autowired        no                                                 
  Autoconfigured   no                                                 
 ---------------- --------------------------------------------------- 
```

```shell
bin/console debug:container router        

 // This service is a public alias for the service Shopware\Storefront\Framework\Routing\Router                         

Information for Service "Shopware\Storefront\Framework\Routing\Router"
======================================================================

 ---------------- ---------------------------------------------- 
  Option           Value                                         
 ---------------- ---------------------------------------------- 
  Service ID       Shopware\Storefront\Framework\Routing\Router  
  Class            Shopware\Storefront\Framework\Routing\Router  
  Tags             -                                             
  Public           no                                            
  Synthetic        no                                            
  Lazy             no                                            
  Shared           yes                                           
  Abstract         no                                            
  Autowired        yes                                           
  Autoconfigured   no                                            
 ---------------- ---------------------------------------------- 
```

**Comment by boxblinkracer** [comment]
LOL,
ok now it makes sense, because my apple pay controller has only the router, and i'm sure it has been tested with multiple channels and urls
so only the other part is wrong. thank you for this
ill create a ticket, check with the dev and revert that change :)
...maybe it was just a bug in shopware one day...i dont know :D

**Comment by AndreasA** [comment]
Yes. As far as I can tell it's the only time the "router.default" is used by the plugin which also suggests that it is wrongly used here.

**Comment by boxblinkracer** [comment]
released with v2.0.0
